coffea/nanoevents/factory.py

```diff
@@ -291,6 +297,12 @@ def from_root(
             elif schemaclass is TreeMakerSchema:
                 from coffea.nanoevents.methods import base, vector
 
+                behavior = {}
+                behavior.update(base.behavior)
+                behavior.update(vector.behavior)
+            elif schemaclass is EDM4HEPSchema:
```

Comment:
This is following the existing pattern, so it's a question for @lgray: why not use the class property `EDM4HEPSchema.behavior` here?

Reply:
This needs cleanup in a separate PR yo harmonize the interface with the eager one. You're right but marking it for later.